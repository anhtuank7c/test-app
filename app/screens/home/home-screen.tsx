import React from 'react';
import {View, Text} from 'react-native';
import {NativeStackScreenProps} from '@react-navigation/native-stack';
import {Button} from '../../components';
import {useAuth} from '../../hooks/useAuth';
import {NavigatorParamsList} from '../../navigators/app-navigator';
import styles from './home-screen.styles';

type Props = NativeStackScreenProps<NavigatorParamsList, 'home'>;
export const HomeScreen = (props: Props) => {
  const {onLogout} = useAuth({navigation: props.navigation});
  return (
    <View style={styles.container}>
      <Text testID="header" style={styles.header}>
        Welcome to demo app
      </Text>
      <Text testID="subtitle" style={styles.subtitle}>
        Test your code, not your patience
      </Text>
      <Button testID="logoutBtn" color="red" text="Logout" onPress={onLogout} />
    </View>
  );
};
