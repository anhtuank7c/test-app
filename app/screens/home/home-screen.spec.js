import React from 'react';
import {render, screen} from '@testing-library/react-native';
import {HomeScreen} from './home-screen';

/**
 * Test View
 */
test('render correctly', () => {
  const {getByTestId} = render(<HomeScreen />);
  expect(getByTestId('header')).toBeTruthy();
  expect(getByTestId('subtitle')).toBeTruthy();
  expect(getByTestId('logoutBtn')).toBeTruthy();
});

test('match snapshot', () => {
  render(<HomeScreen />);
  expect(screen.toJSON()).toMatchSnapshot();
});
