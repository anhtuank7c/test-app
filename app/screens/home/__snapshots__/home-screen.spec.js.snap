// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<View
  style={
    Object {
      "padding": 16,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 27,
        "fontWeight": "bold",
        "letterSpacing": 1,
      }
    }
    testID="header"
  >
    Welcome to demo app
  </Text>
  <Text
    style={
      Object {
        "fontSize": 17,
        "marginBottom": 26,
        "marginTop": 10,
      }
    }
    testID="subtitle"
  >
    Test your code, not your patient
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "red",
        "borderRadius": 10,
        "marginTop": 26,
        "opacity": 1,
        "paddingVertical": 16,
      }
    }
    testID="logoutBtn"
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontSize": 17,
          "letterSpacing": 0.2,
        }
      }
    >
      Logout me out
    </Text>
  </View>
</View>
`;
